client/lib/load-script/callback-handler.js

```diff
@@ -0,0 +1,90 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+const debug = debugFactory( 'lib/load-script/callback-handler' );
+
+/**
+ * Module variables
+ */
+const callbacksForURLsInProgress = new Map();
+
+export function getCallbackMap() {
+	return callbacksForURLsInProgress;
+}
+
+export function isLoading( url ) {
+	return callbacksForURLsInProgress.has( url );
+}
+
+export function addScriptCallback( url, callback ) {
+	if ( isLoading( url ) ) {
+		debug( `Adding a callback for an existing script from "${ url }"` );
+		callbacksForURLsInProgress.get( url ).push( callback );
+	} else {
+		debug( `Adding a callback for a new script from "${ url }"` );
+		callbacksForURLsInProgress.set( url, [ callback ] );
+	}
+}
+
+export function removeScriptCallback( url, callback ) {
+	debug( `Removing a known callback for a script from "${ url }"` );
+
+	if ( ! callbacksForURLsInProgress.has( url ) ) {
+		return;
+	}
+
+	const index = callbacksForURLsInProgress.get( url ).indexOf( callback );
+
+	if ( -1 === index ) {
+		return;
+	}
+
+	if ( 1 === callbacksForURLsInProgress.get( url ).length ) {
+		callbacksForURLsInProgress.delete( url );
+		return;
+	}
+
+	callbacksForURLsInProgress.get( url ).splice( index, 1 );
```

Comment:
Looks like we're making multiple calls to get the callbacks. Could we help ourselves by "caching" it at the start of the function call and reusing our local value?

```js
const urlCallbacks = getCallbacks( url );

if ( ! includes( urlCallbacks, callback ) ) { … }

if ( 1 === urlCallbacks.length ) { … }

return urlCallbacks.splice( … );
```